<launch>
    <!-- Parameters -->
    <arg name="setup_file"  default="lsd_slam_euroc_mono_imu.launch" />
    <arg name="pose_topic"  default="/lsd_slam/pose" />
    <arg name="msg_type"     default="PoseStamped" />
    <arg name="output_file" default="$(find rudpt_evaluation)/eval_data/test.txt" />
    
    <!-- Setup the lsd_slam environment with the given setup file -->
    <!-- <include file="$(find rudpt_evaluation)/launch/lsd_slam/$(arg setup_file)" /> -->
    <include file="$(find lsd_slam_core)/launch/euroc.launch" />
    
    <!-- Initialise node for saving the lsd_slam poses -->
    <node name="path_writer_node" pkg="rudpt_evaluation" type="path_writer_node.py" output="screen" clear_params="true">
        <param name="pose_topic" value="$(arg pose_topic)" type="str" /> 
        <param name="msg_type" value="$(arg msg_type)" type="str" />
        <param name="output_file" value="$(arg output_file)" type="str" />
    </node>

</launch>

